### PR TITLE
Qt: Switch to the C++11 connect() syntax.

### PR DESCRIPTION
**Pull request by waddlesplash**
A few connect()s have not been migrated because the best way to migrate them requires somewhat invasive changes. Other than that, everything has been moved over.

~~The FFmpeg/GIF export changes **have not been tested** and are almost certainly busted because I don't have FFmpeg or ImageMagick here to test, so don't merge until Travis passes.~~ Fixed now.

### CODE REVIEW

**Comment by waddlesplash** [comment]
OK, @endrift, this is ready for review/merge.

File: src/platform/qt/ConfigController.cpp [hunk 1]
```diff
@@ -76,14 +76,12 @@ void ConfigOption::setValue(const char* value) {
 }
 
 void ConfigOption::setValue(const QVariant& value) {
-	QPair<QAction*, QVariant> action;
-	foreach (action, m_actions) {
+	for (QPair<QAction*, QVariant> action : m_actions) {
```
Comment (lioncash) [inline src/platform/qt/ConfigController.cpp]: This should probably be taken by reference.

File: src/platform/qt/ConfigController.cpp [hunk 2]
```diff
@@ -76,14 +76,12 @@ void ConfigOption::setValue(const char* value) {
...
 		bool signalsEnabled = action.first->blockSignals(true);
 		action.first->setChecked(value == action.second);
 		action.first->blockSignals(signalsEnabled);
 	}
-	std::function<void(const QVariant&)> slot;
-	foreach(slot, m_slots.values()) {
+	for (std::function<void(const QVariant&)> slot : m_slots.values()) {
```
Comment (lioncash) [inline src/platform/qt/ConfigController.cpp]: ditto

File: src/platform/qt/AssetTile.cpp [hunk 3]
```diff
@@ -32,7 +32,7 @@ AssetTile::AssetTile(QWidget* parent)
 	m_ui.color->setDimensions(QSize(1, 1));
 	m_ui.color->setSize(50);
 
-	connect(m_ui.preview, SIGNAL(indexPressed(int)), this, SLOT(selectColor(int)));
+	connect(m_ui.preview, &QGBA::Swatch::indexPressed, this, &AssetTile::selectColor);
```
Comment (endrift) [inline src/platform/qt/AssetTile.cpp]: Is there a conflict that is preventing this from just being `Swatch::` instead of `QGBA::Swatch::`?

**Comment by waddlesplash** [comment]
@endrift Fixed.

**Comment by waddlesplash** [comment]
@endrift If you could merge this sooner rather than later, that'd be great, because then I can move on to fixing 0.6.0 issues and the like.

**Pull request merged**